pype/plugins/maya/publish/validate_frame_range.py

```diff
@@ -67,69 +68,32 @@ def process(self, instance):
-                          ))
-
-        if int(animEndTime) != inst_end:
-            errors.append("Animation start in Maya is set to frame [ {} ] "
-                          " and doesn't match the one set on asset [ {} ]: "
-                          "{}/{}/{}/{} (handle/start/end/handle)".format(
-                              animEndTime,
-                              frame_end_handle,
-                              handle_start, frame_start, frame_end, handle_end
-                          ))
-
-        render_start = int(cmds.getAttr("defaultRenderGlobals.startFrame"))
-        render_end = int(cmds.getAttr("defaultRenderGlobals.endFrame"))
-
-        if int(render_start) != inst_start:
-            errors.append("Render settings start frame is set to [ {} ] "
-                          " and doesn't match the one set on asset [ {} ]: "
-                          "{}/{}/{}/{} (handle/start/end/handle)".format(
-                              int(render_start),
-                              frame_start_handle,
-                              handle_start, frame_start, frame_end, handle_end
-                          ))
-
-        if int(render_end) != inst_end:
-            errors.append("Render settings end frame is set to [ {} ] "
-                          " and doesn't match the one set on asset [ {} ]: "
-                          "{}/{}/{}/{} (handle/start/end/handle)".format(
-                              int(render_end),
-                              frame_end_handle,
-                              handle_start, frame_start, frame_end, handle_end
-                          ))
+        if "renderlayer" in self.families:
+
+            render_start = int(cmds.getAttr("defaultRenderGlobals.startFrame"))
+            render_end = int(cmds.getAttr("defaultRenderGlobals.endFrame"))
+
+            if int(render_start) != inst_start:
+                errors.append("Render settings start frame is set to [ {} ] "
+                              "and doesn't match the one set on "
+                              "asset [ {} ]: "
+                              "{}/{}/{}/{} (handle/start/end/handle)".format(
+                                  int(render_start),
+                                  frame_start_handle,
+                                  handle_start, frame_start, frame_end,
+                                  handle_end
+                              ))
+
+            if int(render_end) != inst_end:
+                errors.append("Render settings end frame is set to [ {} ] "
+                              "and doesn't match the one set on "
+                              "asset [ {} ]: "
+                              "{}/{}/{}/{} (handle/start/end/handle)".format(
+                                  int(render_end),
+                                  frame_end_handle,
+                                  handle_start, frame_start, frame_end,
+                                  handle_end
+                              ))
```

Comment:
This simple check for the start and end frame will be wrong if a renderlayer has a Renderlayer override. Technically this would be redundant to validate anyway, since for renderlayers the `instance.data["startFrame"]` is based on exactly those start and end frame of the render settings. So this render settings start/end frame should be removed and one should only rely on validating the actual start/end frame (and of course the handles on the instance if needed).

The start and end frame for renderlayers are collected [here](https://github.com/pypeclub/pype/blob/80aa8a52b755b36a11769d2a4e81ee7fb3b189e8/pype/plugins/maya/publish/collect_render.py#L206-L209).

Reply:
I've updated validator and its repair function to work only on instastance data.